src/entities/purchase-params.ts

```diff
@@ -25,6 +25,12 @@ export interface PurchaseParams {
    */
   customerEmail?: string;
 
+  /**
+   * The step ID from the workflow where the purchase is being initiated.
+   * @internal
+   */
+  stepId?: string;
```

Comment:
Just to confirm, this is not known at SDK configuration time, so we can't include this info in the `WorkflowContext` there right? 

I'm not in love with splitting the information given to the SDK between configuration time and purchase time like this, specially when it means modifying the API like this. I'm not totally familiar with the architecture on how workflows will work though... And it's internal so I guess it's ok for now...

Maybe we could consider passing something like an object `WorkflowPurchaseContext` however, to group this information together, in case we want to add more data in the future, to not have too many generic fields like this?

Another option could be to use the `metadata` field below, but I'm not sure if that could be problematic in the backend.